src/main/java/com/codegym/finwallet/service/impl/TransactionServiceImpl.java

```diff
@@ -405,4 +407,33 @@ private void sendScheduledEmails(List<String> emailList, String subject, DateRes
     private Profile getProfileByEmail(String email){
         return profileRepository.findProfileByEmail(email).orElse(null);
     }
+
+    private void updateAmountWallet(TransactionRequest request, Transaction transaction){
+        Wallet wallet = transaction.getWallet();
+        double requestAmount = request.getAmount();
+        double transactionAmount = transaction.getAmount();
+        double currentAmount = wallet.getAmount();
+        double newAmount = 0;
+        if (requestAmount > transactionAmount) {
+            newAmount = requestAmount - transactionAmount;
+            currentAmount -= newAmount;
+        }else {
+            newAmount = transactionAmount - requestAmount;
+            currentAmount += newAmount;
+        }
+        wallet.setAmount(currentAmount);
+        walletRepository.save(wallet);
+    }
+
+    private boolean isRequestAmountAvailable(TransactionRequest request, Transaction transaction){
+        Wallet wallet = transaction.getWallet();
+        double walletAmount = wallet.getAmount();
+        double transactionAmount = transaction.getAmount();
+        double requestAmount = request.getAmount();
+        double totalAmount = walletAmount + transactionAmount;
+        if (requestAmount > totalAmount){
```

Comment:
should use conditional operator instead of if-else

Reply:
resolved